### DISCUSSION TITLE
Usando Async/Await babel em produção

### DISCUSSION THREAD

**Discussion by RatoX** [frontendbr/forum#474] [Perguntas e Respostas]
O novo padrão de processamento[ síncrono tá chegando!!](https://github.com/tc39/ecmascript-asyncawait) porém já conseguimos ter essa função usando o Babel eu fiquei muito tentado com a legibilidade do código, adeus Callback Hell, e comecei a usar.

O que vocês acham? Vale a pena manter ou é melhor refatorar para as promise nativas mesmo?

**Comment by fdaciuk** [comment]
Eu gosto do `async/await`, pra várias casos ele cai como uma luva.

O interessante dele é que ele não depende diretamente de `Promise`, apesar desse ser o uso mais comum.

O que acredito que tem que tomar mais cuidado é de não ficar enchendo seu código de `try/catch`, destruindo a legibilidade e, muitas vezes, escondendo bugs que vão ser difíceis de encontrar no futuro =)

Mas sobre usar hoje, não vejo problemas =)

**Comment by kazzkiq** [comment]
@fdaciuk Uso `async/await` com o seu projeto [**ajax**](https://github.com/fdaciuk/ajax) e funciona **lindamente**! Melhor library pra requisições AJAX já feita na história moderna! 😄 

Se com Promises as requisições AJAX (e outras funções assíncronas) já ficam muito mais elegantes de escrever, com `async/await` então da até gosto de disparar requests. Código menor, mais legível e muito mais elegante.

Só é importante atentar para questões de compatibilidade com navegadores antigos, *error handling* adequado, e algumas "particularidades" que você encontra ao usar esse tipo de função (como o fato não ser possível utilizar estas funções corretamente dentro de um loop `.forEach()`, [por exemplo](http://stackoverflow.com/questions/37576685/using-async-await-with-a-foreach-loop)).
